lib/vector-tap/src/lib.rs

```diff
@@ -182,6 +182,9 @@ impl<'a> TapRunner<'a> {
                                 )
                             })
                             .collect();
+                        if output_events.is_empty() {
```

Comment:
Rather than this, would it be possible for the channel receiver to be able to handle an empty list of events? Just wondering if you'd considered and discarded that option already.

Reply:
The issue is the buffer size of the channel, if it's too small then the async channel output type will have issues. Though now that I'm thinking about it, this might not be the ideal solution. Let me do some further investigation